### DISCUSSION TITLE
Merkle proof hashes order

### DISCUSSION THREAD

**Discussion by june-cho-88** [gcash/bchd#482] [Q&A] [answered]
I'm trying to validate the proof using gRPC.

When I get an array of merkle proof hashes, it is ordered left to right on the entire merkle tree.
Is there any specific reason for this?
To use given flag bits which are ordered child node to parent node, I have to rearrange the order of hashes. This step doesn't make any sense in the validation process.

What is a benefit of current order structure?

**Comment by zquestz** [comment, marked as answer]
This was inherited from btcd. I am happy to make adjustments so this makes more sense. =)
